proposals/0000-extract-runtime.md

```diff
@@ -0,0 +1,419 @@
+points to make it elastic enough to be extended by UI layer or platform layer, if needed.
+
+This includes, but is not limited to:
+
+ - Event Loop
+ - JS engine management
+ - C++-only Turbo Module hosting
+
+And ideally, in the future (out of scope for this RFC):
+ - DevTools interface and debugging host (long term, after the interface gets stabilized)
+ - Handling of Shared and Dedicated Workers
+
+`react-native` would then focus mostly on:
+
+  - Shadow Nodes
+  - Rendering
+  - UI TurboModules (?)
+
+The parts of the React Native that are not stable enough or are just being tested out
+can be kept in `react-native` package until becoming more stable, with less frequent breaking changes
+and then moved to `react-native-runtime` with more stable interface.
+
+---
+
+Having a runtime layer extract some of the pieces to shared layer:
+
+![Diagram with Runtime](./assets/0000-diagram-with-runtime.png)
+
+Now, with React Native layer focusing solely on UI layer, having an additional OOT platform would make the former diagram 
+look more like the following (with additional APIs the user decided to add):
+
+![Diagram with Runtime and OOT](./assets/0000-diagram-runtime-multiplatform.png)
+
+The new layer would take care of `jsi` interface surface and let React Native (core/UI)
+focus on working with UI and executing the user code, no matter the engine is currently selected.
+
+At the heart of `react-native-runtime` would be the event-loop and JS interface, on which other building blocks 
+would be based on.
+
+This would, for example, allow community to implement custom storage solution using `localStorage` API,
+ideally matching the DOM spec. By having the DevTools Interface as part of this package, they can also
+hook up into `Storage` tab from `Dev tools` to provide excellent Developer Experience.
+
+While the `react-native-runtime` would need to maintain those generic interfaces, the React Native community
+can contribute with implementations of such interfaces (JS, EventLoop and DevTools).
+
+The same can be done with e.g. networking, where C++-modules would be able to hook up to `Networking`
+tab to report and display network requests to provide the best DX possible. This would greatly benefit
+brownfield applications where most of the code is still native.
+
+### Building tools
+
+The Runtime and its turbo modules could be built using CMake only, as they are all written in C++,
+leaving room for React Native or perhaps even React Native Frameworks to use any other build tool that
+is in any way compatible with CMake.
+
+For example, an example application project graph may look like following:
+
+```
+my-rn-app
+├── react-native-runtime
+│   ├── lib-rnrt-fetch
+│   ├── lib-rnrt-crypto
+│   ├── lib-rnrt-mmkv
+│   │   ├── android
+│   │   ├── ios
+│   │   └── windows
+│   └── libhermes
+├── react-native
+│   ├── react-native-gesture-handler
+│   ├── react-native-maps
```

Comment:
I don't understand this. It reads as "react-native-maps is a dependency of react-native" which I'm sure is not what you meant

Reply:
This shows that it depends on `react-native`, and above libs depend on `react-native-runtime` instead (build and linking wise)